include/boost/di.hpp

```diff
@@ -20,6 +20,9 @@
 #if !defined(BOOST_DI_CFG_DIAGNOSTICS_LEVEL)
 #define BOOST_DI_CFG_DIAGNOSTICS_LEVEL 1
 #endif
+#if !defined(BOOST_DI_THREAD_SAFE)
```

Comment:
hmm, not sure of putting it into boost/di.hpp yet as it's just used by the extensions. What about defining it as BOOST_DI_NOT_THREAD_SAFE, this way we don't have to define it all for thread safety?

Reply:
You are right, but in the nearest future we will have to have this flag for shared singleton and it is explicitly self-documented while BOOST_DI_NOT_THREAD_SAFE is not obvious